homeassistant/components/control4/media_player.py

```diff
@@ -0,0 +1,407 @@
+        """Fetch data from Control4 director."""
+        try:
+            return await update_variables_for_config_entry(
+                hass, entry, VARIABLES_OF_INTEREST
+            )
+        except C4Exception as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+    coordinator = DataUpdateCoordinator[dict[int, dict[str, Any]]](
+        hass,
+        _LOGGER,
+        name="room",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=scan_interval),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    items_by_id = {
+        item["id"]: item
+        for item in hass.data[DOMAIN][entry.entry_id][CONF_DIRECTOR_ALL_ITEMS]
+    }
+    item_to_parent_map = {
+        k: item["parentId"]
+        for k, item in items_by_id.items()
+        if "parentId" in item and k > 1
+    }
+
+    ui_config = entry_data[CONF_UI_CONFIGURATION]
+
+    entity_list = []
+    for room in all_rooms:
+        room_id = room["id"]
+
+        sources: dict[int, _RoomSource] = {}
+        for exp in ui_config["experiences"]:
+            if room_id == exp["room_id"]:
+                exp_type = exp["type"]
+                if exp_type not in ("listen", "watch"):
+                    continue
+
+                dev_type = (
+                    _SourceType.AUDIO if exp_type == "listen" else _SourceType.VIDEO
+                )
+                for source in exp["sources"]["source"]:
+                    dev_id = source["id"]
+                    name = items_by_id.get(dev_id, {}).get(
+                        "name", f"Unknown Device - {dev_id}"
+                    )
+                    if dev_id in sources:
+                        sources[dev_id].source_type.add(dev_type)
+                    else:
+                        sources[dev_id] = _RoomSource(
+                            source_type={dev_type}, idx=dev_id, name=name
+                        )
+
+        try:
+            hidden = room["roomHidden"]
+            entity_list.append(
+                Control4Room(
+                    entry_data,
+                    coordinator,
+                    room["name"],
+                    room_id,
+                    item_to_parent_map,
+                    sources,
+                    hidden,
+                )
+            )
+        except KeyError:
+            _LOGGER.exception(
+                "Unknown device properties received from Control4: %s",
+                room,
+            )
+            continue
+
+    async_add_entities(entity_list, True)
+
+
+class Control4Room(Control4Entity, MediaPlayerEntity):
+    """Control4 Room entity."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        entry_data: dict,
+        coordinator: DataUpdateCoordinator[dict[int, dict[str, Any]]],
+        name: str,
```

Comment:
I think you should pass `None` here

Reply:
we can' pass none here pylint won't allow. Is it an issue to provide the same name for enitity and device since there is only one entity I think that seems fine.

Reply:
Done